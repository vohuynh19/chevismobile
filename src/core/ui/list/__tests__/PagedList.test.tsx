import {useState} from 'react';

import {Text} from 'common/components/Text';
import {act, fireEvent, render, renderHook, screen} from 'testUtils';

import {PagedList} from '../PagedList';

const list = () => {
  const data = [];
  for (let i = 1; i < 20; i++) {
    data.push({id: i, name: `Item ${i}`});
  }
  return data;
};

function useDataPagedList() {
  const [data, setData] = useState([1, 2, 3, 4, 5]);

  const onUpdateData = () => {
    setData(prev => prev.concat([6, 7, 8, 9, 10]));
  };
  return {data, onUpdateData};
}

describe('PagedList', () => {
  test('should render the correct list data and the first item passed to ‘data’ is visible on the screen', () => {
    const onRefresh = jest.fn();
    const fetchNextPage = jest.fn();

    // default Flatlist render 10 items first
    render(
      <PagedList
        testID="my-PagedList"
        data={list()}
        onEndReached={() => {}}
        renderItem={({item}) => <Text>{item.name}</Text>}
        onRefresh={onRefresh}
        fetchNextPage={fetchNextPage}
        hasNextPage
        isFetchingNextPage
      />,
      {withSnapshot: true},
    );

    const pagedList = screen.getByTestId('my-PagedList');
    expect(pagedList.props.data).toEqual(list());
    const firstItem = screen.getByText('Item 1');
    expect(firstItem).toBeVisible();
  });

  test('should not render items when items are not visible in the screen', () => {
    const onRefresh = jest.fn();
    const fetchNextPage = jest.fn();

    // default Flatlist render 10 items first
    render(
      <PagedList
        testID="my-PagedList"
        data={list()}
        onEndReached={() => {}}
        renderItem={({item}) => <Text>{item.name}</Text>}
        onRefresh={onRefresh}
        fetchNextPage={fetchNextPage}
        hasNextPage
        isFetchingNextPage
      />,
      {withSnapshot: true},
    );

    // Verify that items are not visible
    const lastItem = screen.queryByText('Item 19');
    expect(lastItem).toBeNull();
  });

  test('Should calls handleRefresh on refresh', () => {
    const onRefresh = jest.fn();
    const fetchNextPage = jest.fn();

    render(
      <PagedList
        testID="my-PagedList"
        data={list()}
        onEndReached={() => {}}
        renderItem={({item}) => <Text>{item.name}</Text>}
        onRefresh={onRefresh}
        fetchNextPage={fetchNextPage}
        hasNextPage
        isFetchingNextPage
      />,
      {withSnapshot: true},
    );

    const flatList = screen.getByTestId('my-PagedList');
    fireEvent(flatList, 'onRefresh');
    expect(onRefresh).toHaveBeenCalled();
    expect(onRefresh).toHaveBeenCalledTimes(1);
  });

  test('should verify user behave scroll to bottom, after the user has stopped scrolling the list with their finger will trigger onMomentumScrollEnd and after that will be trigger fetchNextPageMock ', () => {
    const fetchNextPageMock = jest.fn();
    const onRefresh = jest.fn();

    render(
      <PagedList
        data={[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]}
        showsVerticalScrollIndicator={false}
        onEndReachedThreshold={0.5}
        onRefresh={onRefresh}
        fetchNextPage={fetchNextPageMock}
        hasNextPage
        isFetchingNextPage
        testID="page-list"
        renderItem={({item}) => {
          return <Text>{item}</Text>;
        }}
      />,
      {withSnapshot: true},
    );
    const eventData = {
      nativeEvent: {
        contentOffset: {y: 1, x: 0},
        contentSize: {height: 1, width: 100},
        layoutMeasurement: {height: 100, width: 100}, // Dimensions of the device
      },
    };

    const flatList = screen.getByTestId('page-list');
    fireEvent.scroll(flatList, eventData);

    fireEvent(flatList, 'onMomentumScrollEnd');
    // when onMomentumScrollEnd have been call it will trigger fetchNextPageMock
    expect(fetchNextPageMock).toHaveBeenCalledTimes(1);
  });

  test('should Verify after user behave scroll to bottom trigger fetchNextPageMock, add new items to Pagedlist and verify items now are visible on the screen', async () => {
    const {result} = renderHook(() => useDataPagedList());
    const onRefresh = jest.fn();

    const fetchNextPageMock = jest.fn(() => {
      act(() => {
        result.current.onUpdateData();
      });
    });

    const {rerender} = await render(
      <PagedList
        data={result.current.data}
        showsVerticalScrollIndicator={false}
        onEndReachedThreshold={0.5}
        onRefresh={onRefresh}
        fetchNextPage={fetchNextPageMock}
        hasNextPage
        isFetchingNextPage
        testID="page-list"
        renderItem={({item}) => {
          return <Text>{item}</Text>;
        }}
      />,
      {withSnapshot: true},
    );
    const eventData = {
      nativeEvent: {
        contentOffset: {y: 1, x: 0},
        contentSize: {height: 1, width: 100},
        layoutMeasurement: {height: 100, width: 100}, // Dimensions of the device
      },
    };

    const pagedList = screen.getByTestId('page-list');
    expect(pagedList.props.data).toEqual(result.current.data);
    // current data have 5 item
    const firstItem = screen.getByText('1');
    expect(firstItem).toBeVisible();
    // trigger event scroll and fetchNextPageMock to add new item
    const flatList = screen.getByTestId('page-list');
    fireEvent.scroll(flatList, eventData);

    fireEvent(flatList, 'onMomentumScrollEnd');
    rerender(
      <PagedList
        data={result.current.data}
        showsVerticalScrollIndicator={false}
        onEndReachedThreshold={0.5}
        onRefresh={onRefresh}
        fetchNextPage={fetchNextPageMock}
        hasNextPage
        isFetchingNextPage
        testID="page-list"
        renderItem={({item}) => {
          return <Text>{item}</Text>;
        }}
      />,
    );
    // rerender add check new Item visible in screen
    expect(fetchNextPageMock).toHaveBeenCalledTimes(1);
    const lastItem = screen.getByText('9');
    expect(lastItem).toBeVisible();
  });
});
