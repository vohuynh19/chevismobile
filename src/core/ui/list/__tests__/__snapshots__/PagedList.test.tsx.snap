// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagedList Should calls handleRefresh on refresh 1`] = `
<RCTScrollView
  ListFooterComponent={
    <PagedListFooter
      isLoading={true}
    />
  }
  data={
    [
      {
        "id": 1,
        "name": "Item 1",
      },
      {
        "id": 2,
        "name": "Item 2",
      },
      {
        "id": 3,
        "name": "Item 3",
      },
      {
        "id": 4,
        "name": "Item 4",
      },
      {
        "id": 5,
        "name": "Item 5",
      },
      {
        "id": 6,
        "name": "Item 6",
      },
      {
        "id": 7,
        "name": "Item 7",
      },
      {
        "id": 8,
        "name": "Item 8",
      },
      {
        "id": 9,
        "name": "Item 9",
      },
      {
        "id": 10,
        "name": "Item 10",
      },
      {
        "id": 11,
        "name": "Item 11",
      },
      {
        "id": 12,
        "name": "Item 12",
      },
      {
        "id": 13,
        "name": "Item 13",
      },
      {
        "id": 14,
        "name": "Item 14",
      },
      {
        "id": 15,
        "name": "Item 15",
      },
      {
        "id": 16,
        "name": "Item 16",
      },
      {
        "id": 17,
        "name": "Item 17",
      },
      {
        "id": 18,
        "name": "Item 18",
      },
      {
        "id": 19,
        "name": "Item 19",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={0.5}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControl
      onRefresh={
        [MockFunction] {
          "calls": [
            [],
          ],
          "results": [
            {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    />
  }
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={[]}
  testID="my-PagedList"
  viewabilityConfigCallbackPairs={[]}
>
  <RCTRefreshControl />
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 1
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 2
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 3
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 4
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 5
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 6
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 7
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 8
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 9
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 10
      </Text>
    </View>
    <View
      style={
        {
          "height": 0,
        }
      }
    />
    <View
      onLayout={[Function]}
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingVertical": 8,
            },
          ]
        }
      >
        <ActivityIndicator
          animating={true}
          color="#312B78"
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`PagedList should Verify after user behave scroll to bottom trigger fetchNextPageMock, add new items to Pagedlist and verify items now are visible on the screen 1`] = `
<RCTScrollView
  ListFooterComponent={
    <PagedListFooter
      isLoading={true}
    />
  }
  data={
    [
      1,
      2,
      3,
      4,
      5,
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={0.5}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControl
      onRefresh={[MockFunction]}
    />
  }
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={[]}
  testID="page-list"
  viewabilityConfigCallbackPairs={[]}
>
  <RCTRefreshControl />
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        1
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        2
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        3
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        4
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        5
      </Text>
    </View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingVertical": 8,
            },
          ]
        }
      >
        <ActivityIndicator
          animating={true}
          color="#312B78"
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`PagedList should not render items when items are not visible in the screen 1`] = `
<RCTScrollView
  ListFooterComponent={
    <PagedListFooter
      isLoading={true}
    />
  }
  data={
    [
      {
        "id": 1,
        "name": "Item 1",
      },
      {
        "id": 2,
        "name": "Item 2",
      },
      {
        "id": 3,
        "name": "Item 3",
      },
      {
        "id": 4,
        "name": "Item 4",
      },
      {
        "id": 5,
        "name": "Item 5",
      },
      {
        "id": 6,
        "name": "Item 6",
      },
      {
        "id": 7,
        "name": "Item 7",
      },
      {
        "id": 8,
        "name": "Item 8",
      },
      {
        "id": 9,
        "name": "Item 9",
      },
      {
        "id": 10,
        "name": "Item 10",
      },
      {
        "id": 11,
        "name": "Item 11",
      },
      {
        "id": 12,
        "name": "Item 12",
      },
      {
        "id": 13,
        "name": "Item 13",
      },
      {
        "id": 14,
        "name": "Item 14",
      },
      {
        "id": 15,
        "name": "Item 15",
      },
      {
        "id": 16,
        "name": "Item 16",
      },
      {
        "id": 17,
        "name": "Item 17",
      },
      {
        "id": 18,
        "name": "Item 18",
      },
      {
        "id": 19,
        "name": "Item 19",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={0.5}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControl
      onRefresh={[MockFunction]}
    />
  }
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={[]}
  testID="my-PagedList"
  viewabilityConfigCallbackPairs={[]}
>
  <RCTRefreshControl />
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 1
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 2
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 3
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 4
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 5
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 6
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 7
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 8
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 9
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 10
      </Text>
    </View>
    <View
      style={
        {
          "height": 0,
        }
      }
    />
    <View
      onLayout={[Function]}
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingVertical": 8,
            },
          ]
        }
      >
        <ActivityIndicator
          animating={true}
          color="#312B78"
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`PagedList should render the correct list data and the first item passed to ‘data’ is visible on the screen 1`] = `
<RCTScrollView
  ListFooterComponent={
    <PagedListFooter
      isLoading={true}
    />
  }
  data={
    [
      {
        "id": 1,
        "name": "Item 1",
      },
      {
        "id": 2,
        "name": "Item 2",
      },
      {
        "id": 3,
        "name": "Item 3",
      },
      {
        "id": 4,
        "name": "Item 4",
      },
      {
        "id": 5,
        "name": "Item 5",
      },
      {
        "id": 6,
        "name": "Item 6",
      },
      {
        "id": 7,
        "name": "Item 7",
      },
      {
        "id": 8,
        "name": "Item 8",
      },
      {
        "id": 9,
        "name": "Item 9",
      },
      {
        "id": 10,
        "name": "Item 10",
      },
      {
        "id": 11,
        "name": "Item 11",
      },
      {
        "id": 12,
        "name": "Item 12",
      },
      {
        "id": 13,
        "name": "Item 13",
      },
      {
        "id": 14,
        "name": "Item 14",
      },
      {
        "id": 15,
        "name": "Item 15",
      },
      {
        "id": 16,
        "name": "Item 16",
      },
      {
        "id": 17,
        "name": "Item 17",
      },
      {
        "id": 18,
        "name": "Item 18",
      },
      {
        "id": 19,
        "name": "Item 19",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={0.5}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControl
      onRefresh={[MockFunction]}
    />
  }
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={[]}
  testID="my-PagedList"
  viewabilityConfigCallbackPairs={[]}
>
  <RCTRefreshControl />
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 1
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 2
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 3
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 4
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 5
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 6
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 7
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 8
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 9
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        Item 10
      </Text>
    </View>
    <View
      style={
        {
          "height": 0,
        }
      }
    />
    <View
      onLayout={[Function]}
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingVertical": 8,
            },
          ]
        }
      >
        <ActivityIndicator
          animating={true}
          color="#312B78"
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`PagedList should verify user behave scroll to bottom, after the user has stopped scrolling the list with their finger will trigger onMomentumScrollEnd and after that will be trigger fetchNextPageMock  1`] = `
<RCTScrollView
  ListFooterComponent={
    <PagedListFooter
      isLoading={true}
    />
  }
  data={
    [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onEndReached={[Function]}
  onEndReachedThreshold={0.5}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControl
      onRefresh={[MockFunction]}
    />
  }
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={[]}
  testID="page-list"
  viewabilityConfigCallbackPairs={[]}
>
  <RCTRefreshControl />
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        1
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        2
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        3
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        4
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        5
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        6
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        7
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        8
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        9
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          [
            {
              "color": "#211C4E",
              "fontFamily": "System",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 22,
            },
          ]
        }
      >
        10
      </Text>
    </View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingVertical": 8,
            },
          ]
        }
      >
        <ActivityIndicator
          animating={true}
          color="#312B78"
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
